src/components/graph/GraphSearchHistory.tsx

```diff
@@ -0,0 +1,65 @@
+/** *****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ***************************************************************************** */
+
+import * as React from 'react';
+import { observer } from 'mobx-react-lite';
+import { Virtuoso } from 'react-virtuoso';
+import localStorageWorker from '../../util/LocalStorageWorker';
+import { isEventAction } from '../../helpers/event';
+import { BookmarkItem } from '../BookmarksPanel';
+import { useActiveWorkspace } from '../../hooks';
+import { EventMessage } from '../../models/EventMessage';
+import { EventAction } from '../../models/EventAction';
+
+interface Props {
+	history: Array<EventAction | EventMessage>;
+}
+
+const GraphSearchHistory = (props: Props) => {
+	const { history } = props;
+	const activeWorkspace = useActiveWorkspace();
+
+	const computeKey = (item: EventAction | EventMessage) => {
+		return isEventAction(item) ? item.eventId : item.messageId;
+	};
+
+	const renderHistory = React.useCallback(() => {
```

Comment:
Let's not memoize rendering function, observer components are already memoized. Also be careful about dependencies when using useCallback, useMemo. right now `activeWorkspace.onSavedItemSelect` could change but it will not rerender because it's not in dependency list.
Instead we can memoize onRemove function, but that would be useless because currently BookmarkItem isn't memoized. So lets use React.memo in BookmarkItem component

also we can simplify a bit rendering

if (!length) return something

return history items